realm/register/register.gno

```diff
@@ -0,0 +1,57 @@
+package register
+
+import (
+	"std"
+	"strings"
+
+	"gno.land/p/demo/avl"
+)
+
+var players = avl.Tree{} // player -> token
+
+func RegisterPlayer(playerAddr std.Address, token string) {
+	assertIsFaucet()
+
+	if IsAddressRegistered(playerAddr) {
+		panic("faucet already used")
+	}
+	players.Set(string(playerAddr), token)
+
+	sent := std.GetOrigSend()
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+	pkgaddr := std.GetOrigPkgAddr()
+	banker.SendCoins(pkgaddr, playerAddr, sent)
```

Comment:
Just to be sure I understood correctly, so here you are transferring from the realm account to the player account. So I assume the faucet had funded the realm just before ?

Reply:
Il not transferring from the realm account (real); I’m transferring what is still in the pipe, freshly sent by the faucet (unreal), not less, not more.

if ever someone funds the realm without a register, it is just a lost. 

The amount received is directly forwarded in the same transaction; at the end of successful or failed register; the realm balance is the same at the beginning and at the end.